### PR TITLE
Update the docs

### PR DESCRIPTION
**Pull request by abelsiqueira**
- Update to use 0.6
- Docs for NLSModels
- Docs for counters
- Docs for querying functions

### CODE REVIEW

File: docs/src/index.md [hunk 1]
```diff
@@ -42,17 +42,32 @@ Notice that, for the Hessian, the variables ``z^L`` and ``z^U`` are not used.
...
-- an instance of [`NLPModelMeta`](#attributes), which provides information about the problem,
+- an instance of [`NLPModelMeta`](#Attributes-1), which provides information about the problem,
   including the number of variables, constraints, bounds on the variables, etc.
 - other data specific to the provenance of the problem.
 
+## Nonlinear Least Squares
+
+A special type of `NLPModels` are the `NLSModels`, i.e., Nonlinear Least
+Squares models. In these problems, the function ``f(x)`` is given by
+``\Vert F(x)\Vert^2``, where ``F`` is referred as the residual function.
```
Comment (dpo) [inline docs/src/index.md]: Isn't the factor ½ missing here?

**Comment by dpo** [comment]
Looks great, thank you!

**Comment by dpo** [comment]
Oops

**Comment by abelsiqueira** [comment]
Missing 1/2 found :smile:

**Comment by dpo** [comment]
Thanks!

**Pull request merged**